packages/devtools_app/lib/src/memory/flutter/memory_filter.dart

```diff
@@ -0,0 +1,391 @@
+            child: Padding(
+              padding: const EdgeInsets.only(left: 15),
+              child: Column(
+                mainAxisAlignment: MainAxisAlignment.center,
+                crossAxisAlignment: CrossAxisAlignment.start,
+                children: [
+                  Column(
+                    children: [
+                      const TextField(
+                        decoration: InputDecoration(
+                            border: InputBorder.none,
+                            isDense: true,
+                            labelText: 'Filter Snapshot'),
+                      ),
+                      Row(
+                        mainAxisAlignment: MainAxisAlignment.start,
+                        children: [
+                          Checkbox(
+                              value: controller.filterPrivateClasses,
+                              onChanged: (value) {
+                                setState(() {
+                                  controller.filterPrivateClasses = value;
+                                });
+                              }),
+                          const Text('Hide Private Class e.g.,_className'),
+                        ],
+                      ),
+                      Row(
+                        mainAxisAlignment: MainAxisAlignment.start,
+                        children: [
+                          Checkbox(
+                              value: controller.filterZeroInstances,
+                              onChanged: (value) {
+                                setState(() {
+                                  controller.filterZeroInstances = value;
+                                });
+                              }),
+                          const Text('Hide Classes with No Instances'),
+                        ],
+                      ),
+                      Row(
+                        mainAxisAlignment: MainAxisAlignment.start,
+                        children: [
+                          Checkbox(
+                              value: controller.filterLibraryNoInstances,
+                              onChanged: (value) {
+                                setState(() {
+                                  controller.filterLibraryNoInstances = value;
+                                });
+                              }),
+                          const Text('Hide Library with No Instances'),
+                        ],
+                      ),
+                      Row(
+                        mainAxisAlignment: MainAxisAlignment.start,
+                        children: [
+                          const Padding(padding: EdgeInsets.only(top: 30)),
+                          Text('Hide Libraries or Packages '
+                              '(${controller.filteredLibrariesByGroupName.length}):'),
+                        ],
+                      ),
+                      SizedBox(
+                        height: constraints.maxHeight / 4,
```

Comment:
Ah... one of the classic Flutter layout friction points. Fyi @csells, @InMatrix 
I'm guessing you had to add the SizedBox with a custom size to work around an exception placing a ListView within a flex layout.

Is making this listview scroll separately from the rest of the dialog what you want? If not,
`Column(
  children: <Widget>[
    ListView(
      shrinkWrap: true, // use it
    )
  ],
)`
and a SingleChildScrollView wrapping the column might be a better solution/.

Reply:
It is because this list can be very large (list of known libraries 30, 50, etc.) and I don't want to lose the 3 top checkboxes either.